vmdb/app/models/metric/capture.rb

```diff
@@ -110,14 +110,16 @@ def self.perf_capture_timer(zone = nil)
     targets.each do |target|
       interval_name = self.perf_target_to_interval_name(target)
 
-      parent = target.perf_rollup_parent(interval_name)
       options = {}
-      if parent && tasks_by_rollup_parent.has_key?("#{parent.class}:#{parent.id}")
-        pkey = "#{parent.class}:#{parent.id}"
-        tkey = "#{target.class}:#{target.id}"
-        if targets_by_rollup_parent[pkey].include?(tkey)
-          options[:task_id] = tasks_by_rollup_parent[pkey].id
-          options[:force]   = true # Force collection since we've already verified that capture should be done now
+      (target.perf_rollup_parents(interval_name) || []).each do |parent|
+        if tasks_by_rollup_parent.key?("#{parent.class}:#{parent.id}")
+          pkey = "#{parent.class}:#{parent.id}"
+          tkey = "#{target.class}:#{target.id}"
+          if targets_by_rollup_parent[pkey].include?(tkey)
+            # FIXME: check that this is still correct
```

Comment:
I believe this code is specifically for what we call "Cluster realtime collection".  In VMware, there is no such thing as realtime values for Clusters.  However, our alerting code is based off of realtime, so to support alerting on Clusters we need to have realtime values.  The way this works is to aggregate the Host realtime into set of realtime metrics for the Cluster.  It's a little more complicated because the Host realtime collections happen in parallel on the queue, so we use a Task to coordinate that, and when the Host realtime collections complete they callback into the Task.

So, with that, I *think* this should be set to the Cluster and nothing else.  However, since the `tasks_by_rollup_parent` should only have clusters in them anyway, it happens to work out.  If you change the code to just verify each parent in the Hash, it will work itself out.

cc @blomquisg The More You Know
![](http://i.imgur.com/IW8simF.gif)